source/agora/network/Manager.d

```diff
@@ -234,7 +234,11 @@ public class NetworkManager
     protected ConnectionTask[Address] connection_tasks;
 
     /// All connected nodes (Validators & FullNodes)
-    public DList!NetworkClient peers;
+    private DList!NetworkClient peer_list;
+
+    /// Version of peer_list
+    /// incremented to signal PeerRanges that they are invalidated
+    private uint peer_list_version;
```

Comment:
I think you could store a `PeerRange` here directly, couldn't you ? `manager_version` would then just be an internal pointer.

Reply:
> manager_version would then just be an internal pointer.
 
 what do you mean exactly? I feel like `PeerRange` definitely requires pointer to `manager_version` and the `manager_peer_list`.

Reply:
That will probably work but made my head hurt :D